src/GraphODataTemplateWriter/Settings/ConfigurationService.cs

```diff
@@ -74,11 +77,18 @@ private static TemplateWriterSettings LoadSettingsForLanguage()
             return mainTWS;
 
         }
+        /// <summary>
+        /// Cheap "caching" implementation that compromises bettween performance impact, avoiding flaky unit tests because of a shared app domain and not having to refactor this whole settings implementation
+        /// </summary>
+        private static Dictionary<string, TemplateWriterSettings> cheapSettingsCache = new Dictionary<string, TemplateWriterSettings>();
```

Comment:
nit: I think we can drop "cheap" from the name.

btw, great workaround!

Reply:
I forgot that cheap can have a negative connotation in English, what do you think of "simple"?

Reply:
I didn't think about the negative connotation, I thought SettingsCache was descriptive enough as that is the cache we use for the settings at the moment.

What is the additional information we convey with "simple"?

Reply:
that the intention is to keep it simple and this is why it's not implemented with a proper DP.

Reply:
Usually, workarounds tend to stick around for longer than anticipated lifetime. It makes sense to take a note on the implementation detail, which is already covered by the comment and the implementation itself, but it increases the cognitive load slightly while reading all the other references:

- At this line I am checking the simple settings cache
- At this line I am adding to the simple settings cache

vs

- At this line I am checking the settings cache
- At this line I am adding to the settings cache

Variable naming is a very opinionated matter, so I wanted to share my reasoning. It continues to qualify as nitpick ;)

Reply:
thanks for sharing the detailed reasoning.
I'm personally used to longer variable names when they are non-trivial to express the intent, which helps with differentiation and other aspects. But I agree that practice increases the "parsing effort" (it's more verbose).
Let's have @MIchaelMainer be the "innocent and impartial" third party on this one and help us reach quorum :)

Reply:
Because 

> workarounds tend to stick around for longer than anticipated lifetime. 

We call it SettingsCache. If changes need to be done to it, we'll read the comments then. No need to pass around comment info in the variable name.